doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -91,36 +91,41 @@ Numerals
      sign    : + | -
 
 Strings
-  Strings are delimited by ``"`` (double quote), and enclose a sequence of
-  any characters different from ``"`` or the sequence ``""`` to denote the
-  double quote character. In grammars, the entry for quoted strings is
-  :token:`string`.
-
-  .. productionlist:: coq
-     string : " `characters_opt` "
-     characters_opt : characters_opt character
-                    : empty
+  Strings begin and end with ``"`` (double quote).  Use ``""`` to represent
+  a double quote character within a string.  In the grammar, strings are
+  identified with :production:`string`.
 
 Keywords
-  The following identifiers are reserved, and cannot be used as identifiers::
+  The following character sequences are reserved keywords that cannot be
+  used as identifiers::
 
-    _ as at cofix else end exists exists2 fix for
-    forall fun if IF in let match return
-    SProp Prop Set Type then using where with
+    _ Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop
+    SProp Set Theorem Type Variable as at by cofix discriminated else
+    end exists exists2 fix for forall fun if in lazymatch let match
+    multimatch return then using where with
+
+  Note that plugins may define additional keywords when they are loaded.
 
 Special tokens
-  The following sequences of characters are special tokens::
-
-    ! #[ % & && ( () ) * + ++ , - -> . .( ..
-    / /\ : :: :< := :> ; < <- <-> <: <= <> =
-    => =_D > >-> >= ? ?= @ [ \/ ] ^ { | |-
-    || } ~
-
-  Lexical ambiguities are resolved according to the “longest match”
-  rule: when a sequence of non alphanumerical characters can be
-  decomposed into several different ways, then the first token is the
-  longest possible one (among all tokens defined at this moment), and so
-  on.
+  The set of special
+  tokens defined at any given time can vary because the :cmd:`Notation`
+  command can define new tokens.  A :cmd:`Require` command may load more notation definitions,
+  while the end of a :cmd:`Section` may remove notations.
+
+  Here are some of the character sequences that are special tokens::
```

Comment:
If that's the ones that are defined as special tokens without any notation loaded (and thus all the time), maybe you should say so more clearly.

Reply:
That's what's in the list, except I was reluctant to include all the following very specialized tokens. 

@maximedenes, was it really necessary to make all of these tokens?  An alternative approach would add a lexer token category akin to FIELD (= `.` IDENT), except starting with a `#int63_`.

```
token: #int63_add
token: #int63_addc
token: #int63_addcarryc
token: #int63_addmuldiv
token: #int63_compare
token: #int63_div
token: #int63_div21
token: #int63_diveucl
token: #int63_eq
token: #int63_head0
token: #int63_land
token: #int63_le
token: #int63_lor
token: #int63_lsl
token: #int63_lsr
token: #int63_lt
token: #int63_lxor
token: #int63_mod
token: #int63_mul
token: #int63_mulc
token: #int63_sub
token: #int63_subc
token: #int63_subcarryc
token: #int63_tail0
token: #int63_type
```